### DISCUSSION TITLE
Given name of a model, get its fields

### DISCUSSION THREAD

**Discussion by tlingDeeded** [prisma/prisma#18478] [Q&A] [answered]
I would like to create a method that accepts the name of a model and the name of a field in that model, and dynamically generate prisma queries. Something like this:

```
async countSomething(model: Prisma.ModelName, field: ?, value: string ) {
    let count = await (prisma as any)[model].count({
        where:{
            [field] : value
        }
    })
    return count
}
```

Is there any way for me to type the field, to ensure that the field is always going to be there for the given model?

**Comment by millsp** [comment, marked as answer]
Hey. I think you should be able to do this via a Prisma Client Extension. Extensions also provide a type-level API for reusing and extending existing client types. Since your function inside code cannot sanely be made type-safe in TS, the extension will wrap that and make it type-safe on the surface.

```ts
const prisma = new PrismaClient().$extends({
    model: {
        $allModels: {
            async countSomething<T, K extends string>(
                this: T,
                field: K & keyof Prisma.Payload<T, "findMany">['scalars'],
                value: Prisma.Payload<T, "findMany">['scalars'][K]
            ): Promise<number> {
                const ctx = Prisma.getExtensionContext(this) as any
                return ctx.count({ where: { [field] : value } })
            }
        }
    }
})

prisma.user.countSomething('id', 1) // here inputs are type-checked
```

https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions

**Reply by nurul3101** [reply]
@tlingDeeded Did you had a chance to look at Pierre's answer?

If your query is resolved, please consider marking this as an answer. Please let us know in case you have any further queries.

**Comment by tlingDeeded** [comment]
Thank you. I am currently out of town and have limited internet access.
Will review the answer once I am back.

**Reply by nurul3101** [reply]
Thanks for letting us know

**Reply by nurul3101** [reply]
Let us know in case the issue is resolved.
